config/src/main/java/com/typesafe/config/Config.java

```diff
@@ -794,6 +796,44 @@
     Duration getDuration(String path);
 
     /**
+     * Gets a value as a java.time.Period. If the value is
+     * already a number, then it's taken as days; if it's
+     * a string, it's parsed understanding units suffixes like
+     * "10d" or "5w" as documented in the <a
+     * href="https://github.com/typesafehub/config/blob/master/HOCON.md">the
+     * spec</a>. This method never returns null.
+     *
+     * @since 1.3.0
+     *
+     * @param path
+     *            path expression
+     * @return the period value at the requested path
+     * @throws ConfigException.Missing
+     *             if value is absent or null
+     * @throws ConfigException.WrongType
+     *             if value is not convertible to Long or String
+     * @throws ConfigException.BadValue
+     *             if value cannot be parsed as a number of the given TimeUnit
+     */
+    Period getPeriod(String path);
+
+    /**
+     * Gets a value as a java.time.temporal.TemporalAmount.
+     * This method will first try get get the value as a java.time.Duration, and if unsuccessful,
+     * then as a java.time.Period.
+     * This means that values like "5m" will be parsed as 5 minutes rather than 5 months
```

Comment:
In this case "get a TemporalAmount" to me just means getting either a `Duration` or `Period` but not knowing which. 

I could imagine a framework creator might have some setting where they might want to do `OffsetDateTime.now().plus(maintenanceReminderInterval)` and wouldn't want to constrain the user to a period or duration, but do want to prevent taking a `TemporalAmount` directly.